File: 1º Ano/2º Semestre/Programação Orientada a Objetos/Labs/Lab03/NumberedCard.java
public class NumberedCard extends Card
{
    private int number;
    
    public NumberedCard(int number, Suit suit) {
        super(suit);
        this.number = number > 0 ? number : 0;
    }
    
    public NumberedCard(int number, Suit suit, int value) {
        super(suit,value > 0 ? value : 0);
        this.number = number > 0 ? number : 0;  
    }
    
    @Override
    public int getNumber() {
        return this.number;
    }
    
    public String getName() {
        return this.number == 1 ? "Ás" : String.valueOf(this.number);
    }
    
    //@Override
    public String toString() {
        return getName() + " de " + String.valueOf(getSuit());
    }
    
}

File: 3º Ano/Anual/Programação Visual (PV)/Labs/Lab 1/Lab1/NumberedCard.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Lab1
{
    class NumberedCard : Card
    {
        private int number;  // número da carta (1 a 10)

        public NumberedCard(int number, Suit suit) : base(suit)
        {
            this.number = number;
        }

        public NumberedCard(int number, Suit suit, int value) : base(suit,value)
        {
            this.number = number;
        }
        public NumberedCard() : base(Suit.None)
        {
            this.number = 0;
        }

        public int Number() => this.number;

        public override string Name
        { get { 
            switch (Number())
            {
                case 1:
                    return "ás";
                    break;
                case 2:
                    return "dois";
                    break;
                case 3:
                    return "três";
                    break;
                case 4:
                    return "quatro";
                    break;
                case 5:
                    return "cinco";
                    break;
                case 6:
                    return "seis";
                    break;
                case 7:
                    return "sete";
                    break;
                case 8:
                    return "oito";
                    break;
                case 9:
                    return "nove";
                    break;
                case 10:
                    return "dez";
                    break;
                default:
                    return "";
                    break;
                }
            }
        }

        public override string ToString() => "Carta -> " + Name + " de " + base.GetSuit().ToNameString() + "\n";

    }
}
